pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -56,6 +58,26 @@ func (r *RedisReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
+	// We want to reconcile redis-conf ConfigMap before Deployment
+	// to avoid restart redis pods twice in case of User change ConfigMap.
+	// Annotation GenerationID was added to Pod Template to support Upgrade scenario.
+	// this GenerationID is taken from ConfigMap resourceVersion.
+	// If User change Config Map, Operator will revert it to original one,
+	// but resourceVersion could be changed twice (after user change and after operator).
+	// To avoid this scenario by placing CM reconciliation before Deployment
+	if r.ConfigMap != nil {
+		redisConfigMap := r.ConfigMap(redis)
+		isInternalRedis := r.isInternalRedis()
+		if isInternalRedis {
+			redis_configuration := r.ConfigMap(redis).Data["redis.conf"]
+			redisConfigMap.Data["redis.conf"] = redis_configuration + "\n" + "rename-command REPLICAOF \"\"" + "\n" + "rename-command SLAVEOF \"\"" + "\n"
```

Comment:
add those commands in the main template. This string manipulation is prone to errors.

Reply:
Done. Thank you

Reply:
Hi @eguzki , I checked, but anyway wanted to ask you - I think that  after moving to template`rename-command SLAVEOF` and `rename-command  REPLICAOF` will be for Internal Redis Only. ConfigMap `redis-config` that contains `redis.conf` definitions  is for Internal Redis Only . Hope it's correct. Thank you